src/fsharp/FSComp.txt

```diff
@@ -13,6 +13,8 @@ undefinedNameRecordLabelOrNamespace,"The record label or namespace '%s' is not d
 undefinedNameRecordLabel,"The record label '%s' is not defined"
 undefinedNameTypeParameter,"The type parameter '%s' is not defined"
 undefinedNamePatternDiscriminator,"The pattern discriminator '%s' is not defined"
+missingElseBranch,"You have not supplied the \"else\" case for this expression. If / then is an expression and so it must always return a result of the same type in all cases. This expression was expected to have type '%s' but here has type '%s'."
```

Comment:
Suggest this:

    This expression is missing an 'else' branch. The 'then' branch of this expression has type '%s'. The 'else' branch of an 'if ... then' expression may only be omitted if the 'then' branch has type 'unit'.